runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java

```diff
@@ -147,12 +149,20 @@ private static void serializePartition(final EncoderFactory encoderFactory,
     for (final SerializedPartition<K> partitionToConvert : partitionsToConvert) {
       final K key = partitionToConvert.getKey();
 
-
+      if (partitionToConvert.isOffheap()) {
+        try (DirectByteBufferInputStream byteBufferInputStream =
+               new DirectByteBufferInputStream(partitionToConvert.getBuffer())) {
+          final NonSerializedPartition<K> deserializePartition = deserializePartition(
```

Comment:
Can you reuse code?
L155-L157 and L162-L164

The last argument of `deserializePartition()` is `InputStream`, which both `DirectByteBufferInputStream` and `ByteArrayInputStream` extend.